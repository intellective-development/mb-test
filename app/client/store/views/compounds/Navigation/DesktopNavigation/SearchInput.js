// @flow

import * as React from 'react';

import Search from '../shared/Search';
import { MBDynamicIcon, MBLoader, MBInput } from '../../../elements';
import styles from './BrowseBar.scss';
import './SearchInput.scss';

const SearchInput = () => (
  <div className={styles.cmDBrowseBar_Search_Container}>
    <Search
      renderInput={({ value, onKeyDown, onChange, onFocus, is_loading}) => (
        <div className={styles.cmDBrowseBar_Search_Wrapper}>
          <MBDynamicIcon
            name="search"
            className={styles.cmDBrowseBar_Search_Icon}
            width={26}
            height={26} />
          <MBInput.Input
            id="search"
            placeholder="Search products, categories, brands"
            type="search"
            value={value}
            className={styles.cmDBrowseBar_Search_Input}
            onChange={onChange}
            onKeyDown={onKeyDown}
            onFocus={onFocus}
            autoComplete={'off'} />
          {is_loading && (<MBLoader className={'cmSearchInput__Loader'} />)}
        </div>
      )} />
  </div>
);

export default SearchInput;
