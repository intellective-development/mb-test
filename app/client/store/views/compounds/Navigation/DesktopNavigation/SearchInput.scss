@import "~client/store/views/style/variables";

.cmSearchInput__Loader {
  position: absolute;
  right: $standard-spacing + 20px;
  top: $standard-spacing * 0.5;
}