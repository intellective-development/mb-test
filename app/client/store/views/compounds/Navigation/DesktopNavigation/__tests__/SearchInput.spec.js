
import React from 'react';

import TestProvider from 'store/views/__tests__/utils/TestProvider';
import SearchInput from '../SearchInput';

describe('SearchInput', () => {
  it('renders', () => {
    expect(render(
      <TestProvider initial_state={{}}>
        <SearchInput />
      </TestProvider>
    )).toMatchSnapshot();
  });
});
