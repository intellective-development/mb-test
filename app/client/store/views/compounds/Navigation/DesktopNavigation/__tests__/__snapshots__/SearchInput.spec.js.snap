// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders 1`] = `
<div>
  <div
    class="cmSearch__ClickOutsideTarget"
  >
    <div>
      <div
        class="elMBDynamicIcon"
      >
        <svg
          class=""
          height="26"
          viewBox="0 0 32 32"
          width="26"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#222"
            fill-rule="nonzero"
          >
            <path
              d="M14.81 24.342c-5.145 0-9.31-4.22-9.31-9.42C5.5 9.72 9.665 5.5 14.81 5.5c5.145 0 9.31 4.22 9.31 9.421 0 5.2-4.165 9.421-9.31 9.421zm0-2c4.035 0 7.31-3.32 7.31-7.42 0-4.102-3.275-7.422-7.31-7.422-4.034 0-7.31 3.32-7.31 7.421 0 4.102 3.276 7.421 7.31 7.421z"
            />
            <path
              d="M20.251 14.921h-2c0-1.936-1.543-3.5-3.44-3.5-.833 0-1.617.3-2.237.84l-1.313-1.509a5.388 5.388 0 0 1 3.55-1.33c3.007 0 5.44 2.464 5.44 5.5zM20.331 21.94l1.424-1.406 4.499 4.56-1.424 1.404z"
            />
          </g>
        </svg>
      </div>
      <input
        autocomplete="off"
        class="elMBInput"
        id="search"
        placeholder="Search products, categories, brands"
        type="search"
        value=""
      />
    </div>
    <div
      class="cmAutocompleteDropdown"
    />
  </div>
</div>
`;
